### DISCUSSION TITLE
Is there any difference between `train textcat` of prodigy and spacy's train_textcat.py?

### DISCUSSION THREAD

**Discussion by jdixosnd** [explosion/spaCy#6526] [Help: Best practices]
After annotating the dataset on prodigy I train it using prodigy's train recipe and got 82% F1 score. Later I wanted to do some preprocessing before training the model. so switched to spacy and trained the model on the same dataset and hyperparameters but F1 score falls to 70%. 
I wanted to know that is there anything else that I should keep in mind while training the model with spacy?

I am using following script for training,
https://github.com/explosion/spaCy/blob/master/examples/training/train_textcat.py

**Comment by jdixosnd** [comment]
Also, the version of spacy and base model (en_core_web_md) is same in both the cases.

**Comment by svlandeg** [comment]
There are two main settings I think are important to check: which [textcat architecture](https://spacy.io/api/textcategorizer#architectures) are you using when training with spaCy, and have you set `exclusive_classes` correctly? If your data contains exactly one gold label per instance, set is to `True`, otherwise set it to `False`. If that is correct, you might want to try out the 3 different architectures `ensemble`, `simple_cnn` and `bow`. IIRC, Prodigy uses the default `ensemble`, and it looks like the script you cited, uses `simple_cnn`.

FYI - we're working on getting all these different training scripts & methods consolidated in the upcoming spaCy 3 and the corresponding new Prodigy version!

**Comment by jdixosnd** [comment]
Yes in my case data is mutually exclusive. and thus I have set the flag `exclusive_classes` to `True`. 
Also, The dataset tries to classify data in binary. in prodigy it is marked as either false or true. thus in spacy I tried both the ways. (one label marked as true or false and two different labels) but both of them fail to produce the same result as prodigy.
for model selection(forgot to mention this earlier), I went through the docs of spacy and tried all three models but no luck.

P.S. Eagerly waiting for V3.
Thanks for all your effort.
